parlai/mturk/tasks/image_captions_demo/run.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+from parlai.core.params import ParlaiParser
+from parlai.mturk.core.mturk_manager import StaticMTurkManager
+from parlai.mturk.core.worlds import StaticMTurkTaskWorld
+
+import base64
+import os
+from parlai.mturk.tasks.image_captions_demo.task_config import task_config
+
+from io import BytesIO
+from PIL import Image
+
+
+def load_image(path):
+    return Image.open(path).convert('RGB')
+
+
+def get_image_encoded_from_path(path):
+    img = load_image(path)
+    buffered = BytesIO()
+    img.save(buffered, format='JPEG')
+    encoded = str(base64.b64encode(buffered.getvalue()).decode('ascii'))
+    return encoded
+
+
+def main():
+    """
+    Image captioning to demonstrate the capabilities of the static ParlAI-MTurk
+    interface.
+    """
+    argparser = ParlaiParser(False, False)
+    argparser.add_parlai_data_path()
+    argparser.add_mturk_args()
+    opt = argparser.parse_args()
+    opt['frontend_version'] = 1
+    opt.update(task_config)
+
+    directory_path = os.path.dirname(os.path.abspath(__file__))
+    opt['task'] = os.path.basename(directory_path)
+    if 'data_path' not in opt:
+        opt['data_path'] = os.getcwd() + '/data/' + opt['task']
+    # build_pc_data(opt)
+    mturk_manager = StaticMTurkManager(opt=opt)
+    mturk_manager.setup_server(task_directory_path=directory_path)
+
+    try:
+        mturk_manager.start_new_run()
+
+        mturk_manager.ready_to_accept_workers()
+        mturk_manager.create_hits()
+
+        def check_worker_eligibility(worker):
+            return True
+
+        def assign_worker_roles(workers):
+            for w in workers:
+                w.id = 'worker'
+
+        def run_conversation(mturk_manager, opt, workers):
+            image = '/Users/jju/ParlAI/data/yfcc_images/d06b45856ea94a4a26aa112f79a292e2.jpg'
+            image_data = get_image_encoded_from_path(image)
+            world = StaticMTurkTaskWorld(
+                opt,
+                mturk_agent=workers[0],
+                task_data={'image': image_data, 'word_min': 5, 'word_max': 7},
+            )
+            while not world.episode_done():
+                world.parley()
+
+            world.shutdown()
+
+            return world.prep_save_data(workers)
+
+        mturk_manager.start_task(
+            eligibility_function=check_worker_eligibility,
+            assign_role_function=assign_worker_roles,
+            task_function=run_conversation
+        )
+
+    except BaseException:
```

Comment:
I still don't like BaseException instead of Exception

random suggestion: 

have you considered using withs? https://effbot.org/zone/python-with-statement.htm

```python
with mturk_manager:
    pass
```
which would have:
```python
def __exit__(self, type, value, tradeback):
   self.expire_all_unassigned_hits()
   self.shutdown()
```

Just a random thought for future stuff.

Reply:
(at the very least, it seems like you should do a try-finally instead of a try-catch-finally)

Reply:
In this case, `BaseException` is required to catch `KeyboardInterrupt` and others. If the `with` syntax is able to manage that I can switch to it, but I just hadn't tried that kind of syntax before. 

To some degree this discussion belongs elsewhere though as this run file is going to be cut down quite a lot in v1 when it's finally committed.